src/endianity.rs

```diff
@@ -11,103 +12,127 @@ use parser::{Error, Result};
+    ///
+    /// # Panics
+    ///
+    /// Panics when `buf.len() < 2`.
+    #[inline]
+    fn read_u16(self, buf: &[u8]) -> u16 {
+        if self.is_big_endian() {
+            byteorder::BigEndian::read_u16(buf)
+        } else {
+            byteorder::LittleEndian::read_u16(buf)
+        }
     }
-}
 
-impl byteorder::ByteOrder for LittleEndian {
-    fn read_u16(buf: &[u8]) -> u16 {
-        byteorder::LittleEndian::read_u16(buf)
-    }
-    fn read_u32(buf: &[u8]) -> u32 {
-        byteorder::LittleEndian::read_u32(buf)
+    /// Reads an unsigned 32 bit integer from `buf`.
+    ///
+    /// # Panics
+    ///
+    /// Panics when `buf.len() < 4`.
+    fn read_u32(self, buf: &[u8]) -> u32 {
+        if self.is_big_endian() {
+            byteorder::BigEndian::read_u32(buf)
+        } else {
+            byteorder::LittleEndian::read_u32(buf)
+        }
     }
-    fn read_u64(buf: &[u8]) -> u64 {
-        byteorder::LittleEndian::read_u64(buf)
+
+    /// Reads an unsigned 64 bit integer from `buf`.
+    ///
+    /// # Panics
+    ///
+    /// Panics when `buf.len() < 8`.
+    fn read_u64(self, buf: &[u8]) -> u64 {
+        if self.is_big_endian() {
+            byteorder::BigEndian::read_u64(buf)
+        } else {
+            byteorder::LittleEndian::read_u64(buf)
+        }
     }
-    fn read_uint(buf: &[u8], nbytes: usize) -> u64 {
-        byteorder::LittleEndian::read_uint(buf, nbytes)
+
+    /// Reads a signed 16 bit integer from `buf`.
+    ///
+    /// # Panics
+    ///
+    /// Panics when `buf.len() < 2`.
+    #[inline]
+    fn read_i16(self, buf: &[u8]) -> i16 {
+        self.read_u16(buf) as i16
     }
-    fn write_u16(buf: &mut [u8], n: u16) {
-        byteorder::LittleEndian::write_u16(buf, n)
+
+    /// Reads a signed 32 bit integer from `buf`.
+    ///
+    /// # Panics
+    ///
+    /// Panics when `buf.len() < 4`.
+    #[inline]
+    fn read_i32(self, buf: &[u8]) -> i32 {
+        self.read_u32(buf) as i32
     }
-    fn write_u32(buf: &mut [u8], n: u32) {
-        byteorder::LittleEndian::write_u32(buf, n)
+
+    /// Reads a signed 64 bit integer from `buf`.
+    ///
+    /// # Panics
+    ///
+    /// Panics when `buf.len() < 8`.
+    #[inline]
+    fn read_i64(self, buf: &[u8]) -> i64 {
+        self.read_u64(buf) as i64
     }
-    fn write_u64(buf: &mut [u8], n: u64) {
-        byteorder::LittleEndian::write_u64(buf, n)
+
+    /// Writes an unsigned 64 bit integer `n` to `buf`.
+    ///
+    /// # Panics
+    ///
+    /// Panics when `buf.len() < 8`.
+    fn write_u64(self, buf: &mut [u8], n: u64) {
+        if self.is_big_endian() {
+            byteorder::BigEndian::write_u64(buf, n)
+        } else {
+            byteorder::LittleEndian::write_u64(buf, n)
+        }
     }
-    fn write_uint(buf: &mut [u8], n: u64, nbytes: usize) {
-        byteorder::LittleEndian::write_uint(buf, n, nbytes)
+}
+
+/// Little endian byte order.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct LittleEndian();
```

Comment:
Is there a way to avoid needing to create `NativeEndian` values as `NativeEndian {}`?